src/Executor/Promise/Promise.php

```diff
@@ -30,6 +32,19 @@ public function __construct($adoptedPromise, PromiseAdapter $adapter)
         $this->adapter = $adapter;
     }
 
+    /**
+     * @template TFulfilled of mixed
+     * @template TRejected of mixed
```

Comment:
```suggestion
     * @template TFulfilled
     * @template TRejected
```

Should be equivalent, the template types here are unconstrained.